content/blog/0012-effect-3.0.mdx

```diff
@@ -0,0 +1,328 @@
+
+### Effect code
+
+By using Effect the above mess becomes just 25 lines of code (imports & formatting included):
+
+```ts
+import { HttpClient } from "@effect/platform"
+import type { Cause } from "effect"
+import { Effect, Schedule } from "effect"
+
+export const getTodo: (
+  id: number
+) => Effect.Effect<
+  unknown,
+  | HttpClient.error.HttpClientError
+  | Cause.TimeoutException
+> = (
+  id: number
+) =>
+  HttpClient.request.get(`/todos/${id}`).pipe(
+    HttpClient.client.fetchOk(),
+    HttpClient.response.json,
+    Effect.timeout("1 seconds"),
+    Effect.retry(
+      Schedule.exponential(1000).pipe(
+        Schedule.compose(Schedule.recurs(3))
+      )
+    ),
+    Effect.withSpan("getTodo", { attributes: { id } })
+  )
+```
+
+If we unpack each piece in its own code block and allow for type inference it becomes even simpler to read and understand:
+
+```ts
+import { HttpClient } from "@effect/platform"
+import { Effect, Schedule } from "effect"
+
+const retryPolicy = Schedule.exponential(1000).pipe(
+  Schedule.compose(Schedule.recurs(3))
+)
+
+const httpCall = (id: number) =>
+  HttpClient.request.get(`/todos/${id}`).pipe(
+    HttpClient.client.fetchOk(),
+    HttpClient.response.json
+  )
+
+export const getTodo = (id: number) =>
+  httpCall(id).pipe(
+    Effect.timeout("1 seconds"),
+    Effect.retry(retryPolicy),
+    Effect.withSpan("getTodo", { attributes: { id } })
+  )
+```
+
+In this example we can truly see the power of composition, each block of Effect cares about a specific thing (such as schedule definitions, http client, etc) and it does so in such a way that the single pieces compose together.
+
+The last problem we wanted to solve with our code is checking the types at the edge, accounting for that our full example code becomes:
+
+```ts
+import { HttpClient } from "@effect/platform"
+import { Schema } from "@effect/schema"
+import { Effect, Schedule } from "effect"
+
+const retryPolicy = Schedule.exponential(1000).pipe(
+  Schedule.compose(Schedule.recurs(3))
+)
+
+const httpCall = (id: number) =>
+  HttpClient.request.get(`/todos/${id}`).pipe(
+    HttpClient.client.fetchOk(),
+    HttpClient.response.json
+  )
+
+class Todo extends Schema.Class<Todo>("Todo")({
+  id: Schema.number,
+  userId: Schema.number,
+  title: Schema.string,
+  completed: Schema.boolean
+}) {}
+
+export const getTodo = (id: number) =>
+  httpCall(id).pipe(
+    Effect.timeout("1 seconds"),
+    Effect.retry(retryPolicy),
+    Effect.flatMap(Schema.decodeUnknown(Todo)),
+    Effect.withSpan("getTodo", { attributes: { id } })
+  )
+```
+
+## The Future
+
+While the core `effect` library is now stable, the rest of the ecosystem such as `@effect/platform` and `@effect/schema` aren't yet, it will be our first priority to make the ecosystem libraries stable together with adding tons of documentation and examples.
```

Comment:
```suggestion
While the core `effect` library is now stable, the rest of the ecosystem, such as `@effect/platform` and `@effect/schema`, aren't yet. It will be our first priority to make the ecosystem libraries stable, together with adding tons of documentation and examples.
```